contracts/extensions/StreamingPayments.sol

```diff
@@ -0,0 +1,419 @@
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+  }
+
+  /// @notice Called when upgrading the extension
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
+  }
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  /// @notice Creates a new streaming payment
+  /// @param _fundingPermissionDomainId The domain in which the caller holds the funding permission
+  /// @param _fundingChildSkillIndex The index linking the fundingPermissionDomainId to the domainId
+  /// @param _adminPermissionDomainId The domain in which the caller holds the admin permission
+  /// @param _adminChildSkillIndex The index linking the adminPermissionDomainId to the domainId
+  /// @param _domainId The domain out of which the streaming payment will be paid
+  /// @param _startTime The time at which the payment begins paying out
+  /// @param _endTime The time at which the payment ends paying out
+  /// @param _interval The period of time over which _amounts are paid out
+  /// @param _recipient The recipient of the streaming payment
+  /// @param _tokens The tokens to be paid out
+  /// @param _amounts The amounts to be paid out (per _interval of time)
+  function create(
+    uint256 _fundingPermissionDomainId,
+    uint256 _fundingChildSkillIndex,
+    uint256 _adminPermissionDomainId,
+    uint256 _adminChildSkillIndex,
+    uint256 _domainId,
+    uint256 _startTime,
+    uint256 _endTime,
+    uint256 _interval,
+    address payable _recipient,
+    address[] memory _tokens,
+    uint256[] memory _amounts
+  )
+    public
+    notDeprecated
+    validateFundingPermission(_fundingPermissionDomainId, _fundingChildSkillIndex, _domainId)
+    validateAdministrationPermission(_adminPermissionDomainId, _adminChildSkillIndex, _domainId)
+  {
+    uint256 startTime = (_startTime == 0) ? block.timestamp : _startTime;
+
+    require(_tokens.length == _amounts.length, "streaming-payments-bad-input");
+    require(startTime <= _endTime, "streaming-payments-bad-end-time");
+
+    numStreamingPayments++;
+    streamingPayments[numStreamingPayments] = StreamingPayment(_recipient, _domainId, startTime, _endTime, _interval);
+    for (uint256 i; i < _tokens.length; i++) {
+      paymentTokens[numStreamingPayments][_tokens[i]] = PaymentToken(_amounts[i], block.timestamp);
+
+      emit PaymentTokenUpdated(msgSender(), numStreamingPayments, _tokens[i], _amounts[i]);
+    }
+
+    emit StreamingPaymentCreated(msgSender(), numStreamingPayments);
```

Comment:
I think unfortunately this needs to be earlier in the the function (i.e. immediately after creation, before this loop) to avoid issues with the graph if we use it with this extension. If you think about it processing events in order, it'd try to update a token on a streaming payment before it had created it.